autotest/cpp/test_gdal_jpegxr.cpp

```diff
@@ -0,0 +1,389 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Project:  C++ Test Suite for GDAL/OGR
+// Purpose:  Test read/write functionality for JPEG XR format.
+// Author:   Mateusz Loskot <mateusz@loskot.net>
+//
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017, Mateusz Loskot <mateusz@loskot.net>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Library General Public
+// License as published by the Free Software Foundation; either
+// version 2 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Library General Public License for more details.
+//
+// You should have received a copy of the GNU Library General Public
+// License along with this library; if not, write to the
+// Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+// Boston, MA 02111-1307, USA.
+///////////////////////////////////////////////////////////////////////////////
+
+#include <tut.hpp> // TUT
+#include <tut_gdal.h>
+#include <gdal_common.h>
+#include <gdal.h> // GDAL
+#include <gdal_alg.h>
+#include <gdal_priv.h>
+#include <cpl_string.h>
+#include <cstring> // C++
```

Comment:
@schwehr 
> Confusing comment.

Mind you, the `test_gdal_jpegxr.cpp` is copy of any old existing GDAL C++ file.
Those comment used to be used as markers where blocks of headers start. It's dated long time ago, before we started introducing any headers ordering in GDAL. Currently, following recent changes in style, the comment is redundant indeed.

> You mark a C++ style C include as C++? 

It's misinterpretation. The `<cstring>` is not a "C++ style C" include. It is a C++ standard header.